player/playback-engine/src/main/kotlin/com/tidal/sdk/player/playbackengine/offline/crypto/CacheKeyAesCipherDataSourceFactory.kt

```diff
@@ -29,12 +29,12 @@ import androidx.media3.datasource.cache.CacheKeyFactory
  */
 class CacheKeyAesCipherDataSourceFactory(
     private val cacheKeyFactory: CacheKeyFactory,
-    private val secretKey: ByteArray,
+    private val secretKey: ByteArray?,
     private val upstream: DataSource.Factory,
 ) : DataSource.Factory {
 
     override fun createDataSource(): DataSource {
-        val aesCipherDataSource = AesCipherDataSource(secretKey, upstream.createDataSource())
+        val aesCipherDataSource = AesCipherDataSource(secretKey!!, upstream.createDataSource())
```

Comment:
I obviously don't know super much about the context of this, but I'm wondering:
If you force `secretKey` to not be null when using it, why does it have to be nullable in the first place?

If you expect it to have a chance of being null, why force a crash here?
I noticed that before, with `encryption`, there was a similar construction, so I assume there is some constraint forcing this to be this way, but I'd like to understand it better. Can you explain it?

Reply:
Not that I know much more tbh, but my understanding is that if you use Player but intend to not play content that requires what secretKey is for, then you shouldn't be forced to provide a non-null secretKey.

But if this code is called at runtime then you are playing content that does require a secretKey so if you haven't provided one, you've messed up.

There's probably a different way of designing things to allow for the choice without having the nullability, but that's just the way things are done at the moment